File: Data/Entities/PokemonSpecies.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Data.Entities
{
    public class PokemonSpecies
    {
        public int SpeciesID { get; set; }
        public string SpeciesName { get; set; }
        public string SpeciesDescription { get; set; }
        public int PrimaryTypeID { get; set; }
        public int? SecondaryTypeID { get; set; }

        //Navigation Properties
        public Type PrimaryType { get; set; }
        public Type SecondaryType { get; set; }
        public ICollection<Pokemon> Pokemon {get;set;}
    }
}

File: PokedexUI/ClientApp/src/app/models/pokemon-species.ts
import { PokemonType } from './pokemon-type';

export class PokemonSpecies {
  speciesID: number;
  speciesName: string;
  description: string;
  primaryType: PokemonType;
  secondaryType: PokemonType;
}

export class SimpleSpecies {
  speciesID: number;
  speciesName: string;
}
